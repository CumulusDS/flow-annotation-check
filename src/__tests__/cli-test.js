'use strict';

/**
 * @flow
 */

const path = require('path');
const {
  getParser,
  main,
  resolveArgs,
} = require('../cli');

describe('cli', () => {
  describe('resolveArgs', () => {
    const MOCK_ARGS = {
      root: '.',
    };

    it('should parse the root into a full path', () => {
      const result = resolveArgs(MOCK_ARGS);
      expect(result.root).toMatch(/\//);
      expect(result).toMatchObject({
        root: path.resolve(path.join(__dirname, '../..')),
      });
    });

    it('should setup defaults for include', () => {
      const result = resolveArgs(MOCK_ARGS);
      expect(result).toMatchObject({
        include: ['**/*.js'],
      });
    });

    it('should setup defaults for exclude', () => {
      const result = resolveArgs(MOCK_ARGS);
      expect(result).toMatchObject({
        exclude: ['node_modules/**/*.js'],
      });
    });

    it('should let include and exclude pass through when not falsey', () => {
      const result = resolveArgs({
        include: ['*.js'],
        exclude: ['*.coffee'],
        root: '.',
      });
      expect(result).toMatchObject({
        include: ['*.js'],
        exclude: ['*.coffee'],
      });
      expect(result.root).toMatch(/\//);
    });

  });
});
