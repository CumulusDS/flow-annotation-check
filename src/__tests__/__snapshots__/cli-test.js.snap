// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli getParser should print the help message 1`] = `
"usage: index.js [-h] [-v] [-f FLOW_PATH] [-o {text,html-table,csv,junit}]
                [--html-file HTML_FILE] [--csv-file CSV_FILE]
                [--junit-file JUNIT_FILE] [-a] [--allow-weak] [-i INCLUDE]
                [-x EXCLUDE] [--validate]
                [root]

Positional arguments:
  root                  The root directory to glob files from. (default: \`\\".
                        \\"\`)

Optional arguments:
  -h, --help            Show this help message and exit.
  -v, --version         Show program's version number and exit.
  -f FLOW_PATH, --flow-path FLOW_PATH
                        The path to the flow command. (default: \`\\"flow\\"\`)
  -o {text,html-table,csv,junit}, --output {text,html-table,csv,junit}
                        Output format for status/filename pairs. (default: 
                        \`\\"text\\"\`)
  --html-file HTML_FILE
                        Save the html table output directly into HTML_FILE. 
                        (default: \`null\`)
  --csv-file CSV_FILE   Save CSV output directly into CSV_FILE. (default: 
                        \`null\`)
  --junit-file JUNIT_FILE
                        Save jUnit output directly into JUNIT_FILE. (default: 
                        \`null\`)
  -a, --absolute        Report absolute path names. (default: \`false\`)
  --allow-weak          Consider \`@flow weak\` as a accepable annotation. See 
                        https://flowtype.org/docs/existing.html#weak-mode for 
                        reasons why this should only be used temporarily. 
                        (default: \`false\`)
  -i INCLUDE, --include INCLUDE
                        Glob for files to include. Can be set multiple times. 
                        (default: \`[\\"**/*.js\\"]\`)
  -x EXCLUDE, --exclude EXCLUDE
                        Glob for files to exclude. Can be set multiple times. 
                        (default: \`[\\"+(node_modules build flow-typed)/**/*.
                        js\\"]\`)
  --validate            Run in validation mode. This injects errors into 
                        globbed files and checks the flow-annotation status
"
`;
